README.md

```diff
@@ -0,0 +1,211 @@
+
+Returns folder data.
+
+- **responds** with:
+
+  - <span style="color:red;">error</span>, when a user is unauthenticated,
+
+    ```ts
+    HTTP/1.1 403 Forbidden
+    ```
+
+  - <span style="color:red;">error</span>, when a user is authenticated and the requested folder, doesn't exist,
+
+    ```ts
+    HTTP/1.1 404 Not found
+    ```
+
+  - <span style="color:green;">folder data</span> when the user is authenticated and the requested folder exists. A folder structure is always 1 level deep.
+
+    ```ts
+    HTTP/1.1 200 Ok
+
+    {
+        name: 'bar',
+        items: [
+            {
+                type: 'dir',
+                sizeKb: 0,
+                name: string
+            },
+            {
+                type: 'file',
+                sizeKb: number,
+                name: string
+            }
+        ]
+    }
+    ```
+
+#### POST /session
+
+Login.
+
+- **accepts**:
+
+  ```ts
+  {
+      name: string,
+      password: string
+  }
+  ```
+
+- **responds** with:
+
+  - <span style="color:red;">error</span> when credentials don't match any user,
+
+    ```ts
+    HTTP/1.1 400 Bad request
+    ```
+
+  - <span style="color:green;">success</span> code when the session is created. The token cookie is valid for one day.
+
+    ```ts
+    HTTP/1.1 200 Ok
+    Set-Cookie: FOLDER-APP-TOKEN=a3fWa; Expires=Wed, 22 Mar 2022 07:28:00 GMT; SameSite=Strict; Secure; HttpOnly
+    ```
+
+#### DELETE /session
+
+Logout.
+
+- **responds** with:
+
+  - <span style="color:red;">error</span> when session didn't exist
+
+    ```ts
+    HTTP/1.1 400 Bad request
+    ```
+
+  - <span style="color:green;">success</span> code when the session is deleted. Sets session cookie expiration date to past date
+    ```ts
+    HTTP/1.1 200 Ok
+    Set-Cookie: FOLDER-APP-TOKEN=none; Expires=Wed, 15 Mar 2022 07:28:00 GMT; SameSite=Strict; Secure; HttpOnly
+    ```
+
+## Security considerations
+
+All general things we should consider when writing a secure app:
+
+> SAFETY NOTE:
+> Handle exceptions/errors.
+
+> SAFETY NOTE:
+> The server should be protected against Brute-force/DDoS attacks (eg. one host can only use API a few times per minute).
+> I will not implement that.
+
+> SAFETY NOTE:
+> Educate self and other people about security vulnerabilities.
+
+> SAFETY NOTE:
+> If this is for internal usage we should hide it in our secure `VPN`.
+> I will not implement that.
+
+> SAFETY NOTE:
+> The app should be [protected against CSRF](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html).
+> I will try to implement that if I will have time.
+
+> SAFETY NOTE:
+> Folder service should provide `Access-Control-Allow-Origin` only to our frontend origin.
+
+> SAFETY NOTE:
+> Make sure that we cover other [most common security vulnerabilities](https://owasp.org/www-project-top-ten/).
+> I will not implement that.
+
+> SAFETY NOTE:
+> We should write e2e/automated tests to be sure that these cases are covered.
```

Comment:
Yes, plan to add tests.
I decided not to cover all cases only the most important ones (since this is a demonstration):
**on frontend:** 
 - I will check if the authenticated user is able to browse folders
**on backend:**
- if an authenticated user can get folder data
- if an unauthenticated user cannot get folder data

Reply:
info added in 8fca1fbbfc84e7e987c00b6f1eea56791983b510